system-int-test/test/cmr/system_int_test/search/variable/variable_search_test.clj

```diff
@@ -455,3 +455,54 @@
                umm-version/current-version [expected-variable1]
                (search/find-concepts-umm-json
                 :variable {:variable_name updated-variable1-name})))))))))
+
+(deftest variable-search-sort
+  (let [variable1 (variables/ingest-variable-with-attrs {:native-id "var1"
+                                                         :Name "variable"
+                                                         :LongName "Measurement1"
+                                                         :provider-id "PROV2"})
+        variable2 (variables/ingest-variable-with-attrs {:native-id "var2"
+                                                         :Name "Variable 2"
+                                                         :LongName "Measurement2"
+                                                         :provider-id "PROV1"})
+        variable3 (variables/ingest-variable-with-attrs {:native-id "var3"
+                                                         :Name "a variable"
+                                                         :LongName "Long name"
+                                                         :provider-id "PROV1"})
+        variable4 (variables/ingest-variable-with-attrs {:native-id "var4"
+                                                         :Name "variable"
+                                                         :LongName "m.other"
+                                                         :provider-id "PROV1"})]
+    (index/wait-until-indexed)
+
+    (are3 [sort-key expected-variables]
+      (do
+       (variables/assert-variable-search-order
+        expected-variables
+        (variables/search (if sort-key
+                            {:sort-key sort-key}
+                            {}))))
+
+      "Default sort"
+      nil
+      [variable3 variable4 variable1 variable2]
+
+      "Sort by name"
+      "name"
+      [variable3 variable1 variable4 variable2]
```

Comment:
When variable names are the same, the sort order should fall back to the default. Shouldn't variable4 be before variable1? Curious as to how collection sorting behaves in similar situation.

Reply:
Yeah, you're right. I fixed it by adding variable sub-sort. Collections are a bit more complicated when it comes to sorting.